### PR TITLE
rename 'server-to-server sharing' to 'federated cloud'

### PR DESCRIPTION
**Pull request by schiessle**
as discussed

cc @karlitschek @MTRichards

### CODE REVIEW

File: apps/files_sharing/api/server2server.php [hunk 1]
```diff
@@ -34,7 +34,7 @@ class Server2Server {
 	public function createShare($params) {
 
 		if (!$this->isS2SEnabled(true)) {
```
Comment (DeepDiver1975) [inline apps/files_sharing/api/server2server.php]: Please dry-up the code! THX http://de.wikipedia.org/wiki/Don%E2%80%99t_repeat_yourself

Comment (LukasReschke) [inline apps/files_sharing/api/server2server.php]: See also https://github.com/owncloud/core/issues/12454 and the original PR where that was introduced where I pointed that explicitly out. The part with "Custom middlewares are not possible" ;-)

**Comment by karlitschek** [comment]
Great. Thanks :+1:

File: apps/files_sharing/api/server2server.php [hunk 2]
```diff
@@ -34,7 +34,7 @@ class Server2Server {
...
-			return new \OC_OCS_Result(null, 503, 'Server does not support server-to-server sharing');
+			return new \OC_OCS_Result(null, 503, 'Server does not support cloud federation');
```
Comment (MorrisJobke) [inline apps/files_sharing/api/server2server.php]: is it missing the "sharing"?

File: apps/files_sharing/api/server2server.php [hunk 3]
```diff
@@ -152,7 +152,7 @@ public function declineShare($params) {
 	public function unshare($params) {
 
 		if (!$this->isS2SEnabled()) {
-			return new \OC_OCS_Result(null, 503, 'Server does not support server-to-server sharing');
+			return new \OC_OCS_Result(null, 503, 'Server does not support cloud federation');
```
Comment (nickvergessen) [inline apps/files_sharing/api/server2server.php]: I think something like `Server does not support the "Federated Cloud" feature` would be better and more helpful to the users?

**Comment by icewind1991** [comment]
I think the error messages should stay as they are, "cloud federation" tells me nothing as a user but some buzzwords while "server to server sharing" is hard to misunderstand.

A header for the settings it's fine since it's 1) only exposed to admins, 2) the settings themselves still mention sharing.

**Comment by MTRichards** [comment]
In this case, the message should be "Federated Cloud sharing"
instead of Cloud Federation.

The error and the admin settings need to line up so that an admin can solve the problem easily. 
"Aha, a sharing problem! let me go look for the section to enable that"

They should be the same...

File: apps/files_sharing/templates/settings-admin.php [hunk 4]
```diff
@@ -4,7 +4,7 @@
 ?>
 <div class="section" id="fileSharingSettings" >
 
-	<h2><?php p($l->t('Server-to-Server Sharing'));?></h2>
+	<h2><?php p($l->t('Federated Cloud'));?></h2>
```
Comment (nickvergessen) [inline apps/files_sharing/templates/settings-admin.php]: Federated Cloud Sharing here aswell?

**Comment by PVince81** [comment]
@schiesbn can you make the change ?

**Comment by MorrisJobke** [comment]
I fixed it. I'm fine with that change :+1:

**Comment by DeepDiver1975** [comment]
let's ship it then :+1:

**Pull request merged**